src/test/java/com/spotify/folsom/IntegrationTest.java

```diff
@@ -310,13 +310,22 @@ public void testGetKeyNotExist() throws Throwable {
     assertGetKeyNotFound(client.get(KEY1));
   }
 
-  @Test(expected = IllegalArgumentException.class)
-  public void testTooLargeMultiget() throws Exception {
+  @Test
+  public void testPartitionMultiget() throws Exception {
     List<String> keys = Lists.newArrayList();
-    for (int i = 0; i < 1000; i++) {
-      keys.add("key-" + i);
+    for (int i = 0; i < 500; i++) {
+      final String key = "key-" + i;
+      keys.add(key);
+      client.set(key, key, TTL);
```

Comment:
Need to wait for the set to finish if we want the test to be deterministic (but we can at least wait for all of them at the same time if you store them in a list of futures.